<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LBCA</class>
 <widget class="QMainWindow" name="LBCA">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1493</width>
    <height>805</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1493</width>
    <height>805</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1493</width>
    <height>805</height>
   </size>
  </property>
  <property name="sizeIncrement">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>LBCA</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>../../Desktop/image_LBCA_neon_1.jpg</normaloff>../../Desktop/image_LBCA_neon_1.jpg</iconset>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="page_5">
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>820</x>
          <y>530</y>
          <width>651</width>
          <height>231</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>24</pointsize>
         </font>
        </property>
        <property name="text">
         <string>*LBCA(Лбикей) - является аббревиатурой от: Learning the basics of cryptographic algorithms(Обучение основам криптографических алгоритмов)</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>170</x>
          <y>10</y>
          <width>1181</width>
          <height>441</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="font">
            <font>
             <pointsize>48</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="text">
            <string>Добро пожаловать в LBCA</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="font">
            <font>
             <pointsize>37</pointsize>
            </font>
           </property>
           <property name="text">
            <string>В приложении LBCA Вы сможете познакомиться с самыми базовыми алгоритмами шифрования и дешифрования, их принципами работы и историей.</string>
           </property>
           <property name="scaledContents">
            <bool>true</bool>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="buddy">
            <cstring></cstring>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="layoutWidget">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>580</y>
          <width>701</width>
          <height>141</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QLabel" name="label_11">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <pointsize>20</pointsize>
             <italic>false</italic>
             <bold>false</bold>
             <underline>false</underline>
             <strikeout>false</strikeout>
             <stylestrategy>PreferDefault</stylestrategy>
            </font>
           </property>
           <property name="mouseTracking">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Чтобы начать обучение выберите алгоритм шифрования:</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox_3">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>500</width>
             <height>300</height>
            </size>
           </property>
           <property name="sizeIncrement">
            <size>
             <width>15</width>
             <height>15</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>16</pointsize>
            </font>
           </property>
           <property name="editable">
            <bool>false</bool>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <item>
            <property name="text">
             <string>Выберите алгоритм шифрования</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Шифр Цезаря</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Шифр Виженера</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Шифр Плейфера</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QWidget" name="page_6">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <widget class="QTabWidget" name="tabWidget">
          <property name="font">
           <font>
            <pointsize>22</pointsize>
           </font>
          </property>
          <property name="styleSheet">
           <string notr="true">QTabWidget::pane
{
border: 1px;
}
QTabBar::tab
{
background:rgb(2, 0, 57);
color:rgb(242, 250, 255);
}

 QTabBar::tab:selected
{

background: rgb(35, 165, 89);
}</string>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="Shifr_chz">
           <attribute name="title">
            <string>Шифр Цезаря</string>
           </attribute>
           <widget class="QLabel" name="ystr">
            <property name="geometry">
             <rect>
              <x>500</x>
              <y>10</y>
              <width>641</width>
              <height>51</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>Как устроен шифр Цезаря?</string>
            </property>
           </widget>
           <widget class="QLabel" name="prinzh_1">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>50</y>
              <width>1491</width>
              <height>121</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>17</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Принцип работы шифра Цезаря очень прост: каждая буква сдвигается в алфавите на некоторое количество символов, указанное
в ключе. Например, если ключ - 11, то буква А превратится в букву К. Также, если ключ больше чем количество букв
до конца алфавита, то просто  произойдет переход на новый круг. Например, при ключе 11 буква Я станет буквой Й.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="lin">
            <property name="geometry">
             <rect>
              <x>80</x>
              <y>150</y>
              <width>431</width>
              <height>91</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <family>Segoe UI</family>
              <pointsize>18</pointsize>
              <italic>true</italic>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Линейная визуализация шифрования</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="image_czr">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>240</y>
              <width>611</width>
              <height>371</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="../../Desktop/Imagine/resources_from_qt.qrc">:/imagine/Шифр цезаря на русском.jpg</pixmap>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
           <widget class="QLabel" name="label_2">
            <property name="geometry">
             <rect>
              <x>770</x>
              <y>220</y>
              <width>601</width>
              <height>41</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>20</pointsize>
              <italic>false</italic>
              <bold>false</bold>
             </font>
            </property>
            <property name="text">
             <string>Преимущества и недостатки:</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="label">
            <property name="geometry">
             <rect>
              <x>771</x>
              <y>298</y>
              <width>484</width>
              <height>64</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>18</pointsize>
             </font>
            </property>
            <property name="text">
             <string>К преимуществам можно отнести простоту шифрования и дешифрования.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="label_3">
            <property name="geometry">
             <rect>
              <x>771</x>
              <y>394</y>
              <width>480</width>
              <height>186</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>17</pointsize>
             </font>
            </property>
            <property name="text">
             <string>К недостаткам данного метода шифрования можно отнести то, что при таком шифровании сохраняется алфавитный порядок букв. Не маскируется частотность (*) символов. Число ключей ограничено количеством символов в алфавите. </string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="label_4">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>590</y>
              <width>1321</width>
              <height>171</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <family>Sitka Small Semibold</family>
              <pointsize>14</pointsize>
              <italic>false</italic>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>(*) Некоторые буквы в алфавитах встречаются чаще остальных, в следствии этого можно проследить насколько часто встречается тот или иной символ и, зная, какой это алфавит, провести дешифрование данного символа.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="Line" name="line">
            <property name="geometry">
             <rect>
              <x>771</x>
              <y>263</y>
              <width>509</width>
              <height>16</height>
             </rect>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </widget>
          <widget class="QWidget" name="Shdesh">
           <attribute name="title">
            <string>Шифрование/Дешифрование</string>
           </attribute>
           <widget class="QLabel" name="orig_text">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>20</y>
              <width>424</width>
              <height>38</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Введите текст для шифрования </string>
            </property>
           </widget>
           <widget class="QTextEdit" name="textEdit_1">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>70</y>
              <width>541</width>
              <height>201</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
           </widget>
           <widget class="QLabel" name="shrsh_txt">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>300</y>
              <width>571</width>
              <height>38</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Зашифрованный/расшифрованный текст</string>
            </property>
           </widget>
           <widget class="QTextEdit" name="textEdit_2">
            <property name="geometry">
             <rect>
              <x>100</x>
              <y>340</y>
              <width>541</width>
              <height>231</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>90</x>
              <y>610</y>
              <width>531</width>
              <height>91</height>
             </rect>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QPushButton" name="btn_zashifr">
               <property name="font">
                <font>
                 <pointsize>22</pointsize>
                 <bold>true</bold>
                </font>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton:checked {
    background:rgb(2, 0, 57);
    border: 1px solid  rgb(60, 127, 177);
    color:rgb(242, 250, 255);
}</string>
               </property>
               <property name="text">
                <string>Зашифровать</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_rashifr">
               <property name="font">
                <font>
                 <pointsize>22</pointsize>
                 <bold>true</bold>
                </font>
               </property>
               <property name="styleSheet">
                <string notr="true">QPushButton:checked {
    background:rgb(2, 0, 57);
    border: 1px solid  rgb(60, 127, 177);
    color:rgb(242, 250, 255);
}</string>
               </property>
               <property name="text">
                <string>Дешифровать</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
               <property name="autoExclusive">
                <bool>true</bool>
               </property>
               <property name="autoDefault">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QLabel" name="label_6">
            <property name="geometry">
             <rect>
              <x>800</x>
              <y>240</y>
              <width>621</width>
              <height>361</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Ключ состоит из 33 возможных вариантов, потому что шифруются только буквы русского алфавита. Можно отметить интересную особенность, что текст с ключом 0 и 33 полностью совпадает, что и подтверждает суждение о цикличном круговом переходе символов.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QWidget" name="layoutWidget">
            <property name="geometry">
             <rect>
              <x>850</x>
              <y>60</y>
              <width>501</width>
              <height>151</height>
             </rect>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="font">
                <font>
                 <pointsize>22</pointsize>
                </font>
               </property>
               <property name="text">
                <string>Выберите ключ шифрования:</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <widget class="QSlider" name="horizontalSlider">
                 <property name="font">
                  <font>
                   <pointsize>21</pointsize>
                  </font>
                 </property>
                 <property name="maximum">
                  <number>33</number>
                 </property>
                 <property name="orientation">
                  <enum>Qt::Horizontal</enum>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QSpinBox" name="spinBox">
                 <property name="font">
                  <font>
                   <pointsize>22</pointsize>
                  </font>
                 </property>
                 <property name="maximum">
                  <number>33</number>
                 </property>
                 <property name="value">
                  <number>0</number>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </widget>
          <widget class="QWidget" name="hist_sprv">
           <attribute name="title">
            <string>Историческая справка</string>
           </attribute>
           <widget class="QLabel" name="date_czr">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>560</y>
              <width>661</width>
              <height>121</height>
             </rect>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <family>Segoe UI Variable Small Semibol</family>
              <pointsize>22</pointsize>
              <bold>true</bold>
             </font>
            </property>
            <property name="text">
             <string>Гай Юлий Цезарь (100-44 г до н.э.)</string>
            </property>
           </widget>
           <widget class="QLabel" name="info_1">
            <property name="geometry">
             <rect>
              <x>460</x>
              <y>-50</y>
              <width>941</width>
              <height>381</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Шифр Цезаря назван в честь известного древнеримского диктатора Юлия Цезаря. Согласно книге древнеримского историка Светония
“Жизнь двенадцати Цезарей”, он использовал шифр с ключом 3 для защиты военных сообщений. Конечно другие шифры подстановки
использовались и до него, но Цезарь был первым известным пользователем данной схемы.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QPushButton" name="exit_btn">
            <property name="geometry">
             <rect>
              <x>990</x>
              <y>580</y>
              <width>451</width>
              <height>81</height>
             </rect>
            </property>
            <property name="font">
             <font>
              <pointsize>22</pointsize>
             </font>
            </property>
            <property name="text">
             <string>Вернуться в главное меню</string>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
            <property name="autoRepeat">
             <bool>false</bool>
            </property>
            <property name="autoExclusive">
             <bool>false</bool>
            </property>
           </widget>
           <widget class="QLabel" name="Czr">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>441</width>
              <height>561</height>
             </rect>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="pixmap">
             <pixmap resource="../../Desktop/Imagine/resources_from_qt.qrc">:/imagine/Цезарь1.jpg</pixmap>
            </property>
            <property name="scaledContents">
             <bool>true</bool>
            </property>
           </widget>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="../../Desktop/Imagine/resources_from_qt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>766</y>
    </hint>
    <hint type="destinationlabel">
     <x>421</x>
     <y>755</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>horizontalSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>755</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>766</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_3</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>816</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>821</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
